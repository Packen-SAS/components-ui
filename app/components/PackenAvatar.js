import React from "react";

import { ImageBackground } from "react-native";

import * as AvatarStyles from "../styles/components/PackenAvatar";

const PackenAvatar = props => {
  const get_styles = () => {
    return {
      height: AvatarStyles[props.size].size,
      width: AvatarStyles[props.size].size
    };
  }

  const get_src = () => {
    const path = `../../assets/images/${props.src}`;
    return require(path);
  }

  return (
    <ImageBackground source={get_src} style={get_styles}></ImageBackground>
  );
}

export default PackenAvatar;