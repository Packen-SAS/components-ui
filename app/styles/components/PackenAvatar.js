export const tiny = {
  size: 32
};

export const small = {
  size: 40
};

export const medium = {
  size: 64
};

export const large = {
  size: 80
};

export const giant = {
  size: 96
};