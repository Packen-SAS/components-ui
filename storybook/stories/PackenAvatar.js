import React from "react";
import { storiesOf } from "@storybook/react-native";

import Wrapper from "../wrapper";
import PackenAvatar from "../../app/components/PackenAvatar";

storiesOf("PackenAvatar", module)
  .add("Tiny", () => (
    <Wrapper>
      <PackenAvatar
        size="tiny"
        src={require("../../assets/images/avatar.jpg")}
      />
    </Wrapper>
  ))
  .add("Small", () => (
    <Wrapper>
      <PackenAvatar
        size="small"
        src={require("../../assets/images/avatar.jpg")}
      />
    </Wrapper>
  ))
  .add("Medium", () => (
    <Wrapper>
      <PackenAvatar
        size="medium"
        src={require("../../assets/images/avatar.jpg")}
      />
    </Wrapper>
  ))
  .add("Large", () => (
    <Wrapper>
      <PackenAvatar
        size="large"
        src={require("../../assets/images/avatar.jpg")}
      />
    </Wrapper>
  ))
  .add("Giant", () => (
    <Wrapper>
      <PackenAvatar
        size="giant"
        src={require("../../assets/images/avatar.jpg")}
      />
    </Wrapper>
  ));