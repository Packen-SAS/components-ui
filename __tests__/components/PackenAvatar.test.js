import "react-native";
import React from "react";

import PackenAvatar from "../../app/components/PackenAvatar";

import renderer from "react-test-renderer";

describe("<PackenAvatar/>", () => {
  const render = renderer.create(
    <PackenAvatar size="tiny" src={require("../../assets/images/avatar.jpg")}/>
  ).toJSON();

  it("renders correctly", () => {
    expect(render).toMatchSnapshot();
  });
});