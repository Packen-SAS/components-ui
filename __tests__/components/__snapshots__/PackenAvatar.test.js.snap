// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackenAvatar/> renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 100,
      "height": 32,
      "overflow": "hidden",
      "width": 32,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/avatar.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 100,
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
